libp2p/NodeTable.cpp

```diff
@@ -328,50 +315,46 @@ void NodeTable::noteActiveNode(Public const& _pubk, bi::udp::endpoint const& _en
     if (_pubk == m_node.address() || !NodeIPEndpoint(_endpoint.address(), _endpoint.port(), _endpoint.port()).isAllowed())
         return;
 
-    shared_ptr<NodeEntry> node = nodeEntry(_pubk);
-    if (!!node && !node->pending)
+    shared_ptr<NodeEntry> newNode = nodeEntry(_pubk);
+    if (newNode && !newNode->pending)
     {
         clog(NodeTableConnect) << "Noting active node:" << _pubk << _endpoint.address().to_string() << ":" << _endpoint.port();
-        node->endpoint.address = _endpoint.address();
-        node->endpoint.udpPort = _endpoint.port();
-        
-        shared_ptr<NodeEntry> contested;
+        newNode->endpoint.address = _endpoint.address();
+        newNode->endpoint.udpPort = _endpoint.port();
+
+        shared_ptr<NodeEntry> nodeToEvict;
         {
+            // Find a bucket to put a node to
             Guard l(x_state);
-            NodeBucket& s = bucket_UNSAFE(node.get());
-            bool removed = false;
-            s.nodes.remove_if([&node, &removed](weak_ptr<NodeEntry> const& n)
```

Comment:
Iterating over the whole list with `remove_if` was actually not needed, because each node is unique in the bucket.
Now I don't remove at all, but find this node and move it to the back of the list with `splice`